tests/cucumber/test.feature

```diff
@@ -17,7 +17,7 @@ Feature: Example feature
     Scenario: Retry Check
         Then  I should fail once but pass on the second run
 
-    @skip(browserName="chrome")
+    @skip{browserName="chrome"}
```

Comment:
Can you provide information about the reasoning of this change?

Reply:
In the past, we used `parseGherkinMessageStream` beforehand, granting us the ability to filter tests based on tags or custom criteria. This allowed us to manupulate the list of `pickleId` returned to the `cucumber Runtime` for execution.

However, with this new approach, we've transitioned to use the `cucumber API` for test execution, which prevents the prior use of `parseGherkinMessageStream`. Consequently, to skip test cases based on capabilities, we now need a different method by generating a dynamic cucumber tags with a `not` condition to address this challenge.

Moreover, when we encounter tags like `@skip(browser="chrome")`, cucumber's tag expression encounters an issue due to reserved characters `(` and `)`. To overcome this problem within the tag expression, this modification is necessary. As a solution, we generate dynamic tags like `not @skip{browser="chrome"}`, which is compatible with the tag expression.

Reply:
Gotcha. That makes sense. I just thinking how much this will impact users Cucumber tests and whether merging this PR requires a minor or major version bump.

Reply:
It would probably be best to categorise this as a major change, mainly due to the shift in how things are executed underneath.

Reply:
I agree, but this would mean we can't merge it until we work and plan to release WebdriverIO v9. I am not worried about the changes as we have good test coverage and always can fix bugs when reported by users. If we can find a way to keep the old behavior supported I can imagine releasing this as `minor` update. Can you think of a workaround we can keep in place until we release v9?

Reply:
I believe I made an error when escaping special characters. However, with the revised regex, I managed to restore the previous behaviour without causing any impact. Let me update the changes.

Reply:
Hi @christian-bromann, I've made adjustments to the tags, ensuring they are returned with the escape character, and this modification has been accepted by the tag-expression. Thus, from a user standpoint, there shouldn't be any significant changes. Kindly take a look at these revisions and let me know your thoughts.